--- form_ert.C	Mon Aug  6 14:17:17 2001
+++ ../form_ert.C	Mon Aug  6 14:15:39 2001
@@ -8,7 +8,7 @@
 #include FORMS_H_LOCATION
 #include <stdlib.h>
 #include "form_ert.h"
-#include "FormErt.h"
+#include "FormERT.h"
 
 FD_form_ert::~FD_form_ert()
 {
@@ -17,7 +17,7 @@
 }
 
 
-FD_form_ert * FormErt::build_ert()
+FD_form_ert * FormERT::build_ert()
 {
   FL_OBJECT *obj;
   FD_form_ert *fdui = new FD_form_ert;
