--- src/kernel/qapplication_mac.cpp.orig	Thu Jun 19 01:35:24 2003
+++ src/kernel/qapplication_mac.cpp	Thu Jul  3 08:29:41 2003
@@ -1026,6 +1026,7 @@
 { '}', MAP_KEY(Qt::Key_BraceRight) },
 { '|', MAP_KEY(Qt::Key_Bar) },
 { '~', MAP_KEY(Qt::Key_AsciiTilde) },
+{ '^', MAP_KEY(Qt::Key_AsciiCircum) },
 //terminator
 {   0, MAP_KEY(0) } };
 
--- src/kernel/qwindowdefs.h.orig	Wed Jun 11 01:50:44 2003
+++ src/kernel/qwindowdefs.h	Thu Jul  3 08:27:36 2003
@@ -101,7 +101,7 @@
 #ifndef Q_WS_MACX
 typedef struct CGContext *CGContextRef;
 #endif
-typedef struct OpaqueControlRef *ControlRef;
+// typedef struct OpaqueControlRef *ControlRef;
 typedef struct OpaqueWindowGroupRef *WindowGroupRef;
 typedef struct OpaqueGrafPtr *CGrafPtr;
 typedef struct OpaquePMPrintSession *PMPrintSession;
