--- bibforms.h	Tue Oct  7 21:30:55 1997
+++ ../src/bibforms.h	Tue Oct  7 19:57:53 1997
@@ -2,12 +2,6 @@
 #define FD_citation_form_h_
 /* Header file generated with fdesign. */
 
-/**** Callback routines ****/
-
-extern void bibitem_cb(FL_OBJECT *, long);
-
-extern void bibitem_cb(FL_OBJECT *, long);
-
 
 /**** Forms and Objects ****/
 
@@ -19,7 +13,7 @@
 	FL_OBJECT *label;
 } FD_citation_form;
 
-extern FD_citation_form * create_form_citation_form(void);
+
 typedef struct {
 	FL_FORM *bibitem_form;
 	void *vdata;
@@ -27,7 +21,5 @@
 	FL_OBJECT *key;
 	FL_OBJECT *label;
 } FD_bibitem_form;
-
-extern FD_bibitem_form * create_form_bibitem_form(void);
 
 #endif /* FD_citation_form_h_ */
