--- math_forms.c	Fri Dec 12 01:32:42 1997
+++ math_forms.c.patched	Fri Dec 12 01:32:18 1997
@@ -2,7 +2,7 @@
 
 #include "forms.h"
 #include <stdlib.h>
-#include "math_forms.h"
+#include "math_panel.h"
 
 FD_panel *create_form_panel(void)
 {
@@ -17,7 +17,9 @@
   fdui->func_browse = obj = fl_add_browser(FL_SELECT_BROWSER,180,40,94,110,"Functions");
     fl_set_object_lalign(obj,FL_ALIGN_TOP);
     fl_set_object_callback(obj,button_cb,MM_FUNC);
+#if FL_REVISION > 85
     fl_set_browser_hscrollbar(obj, FL_OFF);
+#endif
   fdui->greek = obj = fl_add_button(FL_NORMAL_BUTTON,10,90,50,30,"Greek");
     fl_set_object_lsize(obj,FL_NORMAL_SIZE);
     fl_set_object_lstyle(obj,15);
@@ -79,7 +81,7 @@
 
   fdui->delim = fl_bgn_form(FL_NO_BOX, 250, 260);
   obj = fl_add_box(FL_UP_BOX,0,0,250,260,"");
-  fdui->menu = obj = fl_add_button(FL_PUSH_BUTTON,40,70,170,140,"");
+  fdui->menu = obj = fl_add_bmtable(FL_PUSH_BUTTON,40,70,170,140,"");
     fl_set_object_lcolor(obj,FL_BLUE);
     fl_set_object_callback(obj,delim_cb,2);
 
@@ -166,7 +168,7 @@
 
   fdui->deco = fl_bgn_form(FL_NO_BOX, 160, 150);
   obj = fl_add_box(FL_UP_BOX,0,0,160,150,"");
-  fdui->menu = obj = fl_add_button(FL_PUSH_BUTTON,10,10,140,90,"");
+  fdui->menu = obj = fl_add_bmtable(FL_PUSH_BUTTON,10,10,140,90,"");
     fl_set_object_lcolor(obj,FL_BLUE);
     fl_set_object_callback(obj,deco_cb,MM_APPLY);
   obj = fl_add_button(FL_RETURN_BUTTON,30,110,100,30,"Close");
